common/perf-test-core/src/main/java/com/azure/perf/test/core/MockEventProcessor.java

```diff
@@ -0,0 +1,150 @@
+
+package com.azure.perf.test.core;
+
+import reactor.core.publisher.Mono;
+
+import java.time.Duration;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.function.Consumer;
+import java.util.stream.IntStream;
+
+/**
+ * Represents a Mock event processor.
+ */
+public class MockEventProcessor {
+    private final Consumer<MockErrorContext> processError;
+    private final Consumer<MockEventContext> processEvent;
+    private boolean process;
+    private final double maxEventsPerSecondPerPartition;
+    private final int maxEventsPerSecond;
+    private final int partitions;
+    private final Duration errorAfter;
+    private boolean errorRaised;
+    private final ReentrantLock errorLock;
+
+    private final MockEventContext[] mockEventContexts;
+    private int[] eventsRaised;
+    private long startTime;
+
+    private final AtomicReference<ScheduledFuture<?>> runner = new AtomicReference<>();
+    private final AtomicReference<ScheduledExecutorService> scheduler = new AtomicReference<>();
+
+    /**
+     * Creates an instance of a mock event processor
+     *
+     * @param partitions the number of partitions
+     * @param maxEventsPerSecond the maximum events per second to send, optional.
+     * @param errorAfter the duration after which processor should error out, optional.
+     * @param processError the consumer to process the error.
+     * @param processEvent the consumer to process the event.
+     */
+    public MockEventProcessor(int partitions, int maxEventsPerSecond, Duration errorAfter,
+                              Consumer<MockErrorContext> processError, Consumer<MockEventContext> processEvent) {
+        this.processError = processError;
+        this.processEvent = processEvent;
+        this.partitions = partitions;
+        this.maxEventsPerSecond = maxEventsPerSecond;
+        this.maxEventsPerSecondPerPartition = ((double) maxEventsPerSecond) / partitions;
+        this.errorAfter = errorAfter;
+        this.errorLock = new ReentrantLock();
+
+        mockEventContexts = new MockEventContext[partitions];
+        IntStream.range(0, partitions).boxed().forEach(integer -> {
+            mockEventContexts[integer] = new MockEventContext(integer, "Hello");
+        });
+        this.eventsRaised = new int[partitions];
+    }
+
+    /**
+     * Starts the event processor.
+     */
+    public synchronized void start() {
+        eventsRaised = new int[eventsRaised.length];
+        process = true;
+        errorRaised = false;
+        startTime = System.nanoTime();
+        ScheduledExecutorService executor = Executors.newSingleThreadScheduledExecutor();
+        scheduler.set(executor);
+
+        Double jitterInMillis =
```

Comment:
Why is `jitter` needed?  I don't think the .NET version has this.

Reply:
The Java EH Processors have this, so added to mock one too.
@srnagar likely knows the underlying reason.